src/adapter/src/catalog.rs

```diff
@@ -259,6 +259,23 @@ impl CatalogState {
         }
     }
 
+    /// Indicates whether the indicated item is considered stable or not.
+    ///
+    /// Only stable items can be used as dependencies of other catalog items.
+    pub fn is_stable(&self, id: GlobalId) -> bool {
+        let item = self.get_entry(&id).item();
+
+        match item {
+            CatalogItem::Table(_) => {
+                !(id == self.resolve_builtin_table(&MZ_VIEW_KEYS)
+                    || id == self.resolve_builtin_table(&MZ_VIEW_FOREIGN_KEYS))
```

Comment:
Is there a more efficient way to check if a given `GlobalId` identifies either of these two system tables?

Reply:
No, this is the best way since the IDs are not known at compile time.